ext/ffi/ffi.c

```diff
@@ -984,7 +1030,7 @@ static void zend_ffi_callback_trampoline(ffi_cif* cif, void* ret, void** args, v
 
 	ret_type = ZEND_FFI_TYPE(callback_data->type->func.ret_type);
 	if (ret_type->kind != ZEND_FFI_TYPE_VOID) {
-		zend_ffi_zval_to_cdata(ret, ret_type, &retval);
+		zend_ffi_zval_to_cdata(ret, ret_type, &retval, true);
```

Comment:
Instead of ``zend_ffi_zval_to_cdata()`` modification, we should widen ``ret_type`` to ``intptr_t`` (or whatever else that ABI requests). We should keep the signess of the origianal ``ret_type`` and the capacity of ``ffi_arg``.

Reply:
I got rid of the `ZEND_FFI_WRITE_NARROW` macro in a separate commit to this PR.
I needed to keep the `is_ret` addition to handle cdata objects specially because we must not overread the pointer. I do see the appeal of this approach although I found the `ZEND_FFI_WRITE_NARROW` approach also clear.